### DISCUSSION TITLE
get_all_instruments()

### DISCUSSION THREAD

**Discussion by matinoseda** [itAtPPI/ppi-connector-python#2] [Ideas]
Hola! Me parece buena la idea de crear un método para traer todos los intrumentos que se pueden operar en PPI.

Sería algo similar al método ya existente:
`        instruments = ppi.marketdata.search_instrument("GGAL", "", "Byma", "Acciones")
`

Que te devuelve lo siguiente:
`{'ticker': 'GGAL', 'description': 'Grupo Financiero Galicia', 'currency': 'Pesos', 'type': 'ACCIONES', 'market': 'BYMA'}
{'ticker': 'GGALC', 'description': 'GGALC', 'currency': 'Dolares divisa | CCL', 'type': 'ACCIONES', 'market': 'BYMA'}
{'ticker': 'GGALD', 'description': 'GGALD', 'currency': 'Dolares billete | MEP', 'type': 'ACCIONES', 'market': 'BYMA'}`

Entonces, lo que se podría agregar sería algo así como:
`        instruments = ppi.marketdata.search_all_instruments(ticker="GGAL", name="", market="Byma", acciones="Acciones", settlement="A-48HS")`

Las diferencias entre `ppi.marketdata.search_instrument` y `search_all_instruments` son:
1) que `search_all_instruments` permite , por ejemplo, si no seteás el parámetro *ticker* pero especificás el *market*, que te devuelva todos los instrumentos de ese mercado. 
(La idea del funcionamiento de  `search_all_instruments` es que filtre la búsqueda de instrumentos al especificar los parámetros: `ticker`, `name`, `market`, `acciones`, `settlement` y si no se especifican los parámetros que devuelva todos los intrumentos que no son filtrados). 

2) `search_all_instruments`  también permite filtrar por plazo de CI, 24hs o 48hs con el parámetro `settlement`

**Comment by itAtPPI** [comment]
Buenos días Matias, muchas gracias por el feedback. Por la gran cantidad de instrumentos disponibles hoy en día no permitimos esta operatoria en general pero podriamos llegar a agregarla puntualmente para la api en una proxima version ya que no seria tan pesada la consulta.

**Reply by ferceba00** [reply]
Buenas tardes, muchas gracias de ante mano. Justamente entre hoy a buscar esta info.
Al dia de hoy, no se hizo esa modificacion aun verdad?

mi otra consulta es saber con quien puedo hablar telefonicamente y luego mando mail a api@port... ya que estoy desarrollando una web app para mis seteos   y estrategias pero tambien puede ser algo de esto de interes vuestro.
